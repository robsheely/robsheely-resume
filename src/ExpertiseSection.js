import React from "react";
import PropTypes from "prop-types";

import GridList from "@material-ui/core/GridList";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core";

const styles = theme => ({
  gridList: {
    paddingTop: 10,
    paddingLeft: 20,
    paddingRight: 20,
  },
  bullet: {
    textAlign: "left",
    fontSize: 16
  }
});

const ExpertiseSection = props => {
  const { classes, content } = props;

  return (
    <GridList cellHeight={24} className={classes.gridList} cols={3}>
      {content.map(element => (
        <Typography className={classes.bullet} key={element.id}>â€¢&nbsp;&nbsp;&nbsp;{element.content}</Typography>
      ))}
    </GridList>
  );
};

ExpertiseSection.propTypes = {
  classes: PropTypes.object.isRequired,
  content: PropTypes.array.isRequired
};

export default withStyles(styles)(ExpertiseSection);
