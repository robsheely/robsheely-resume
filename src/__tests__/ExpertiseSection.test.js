import React from "react";
import { render, cleanup } from "react-testing-library";

import ExpertiseSection from "../ExpertiseSection";

const MOCK_DATA = {
  items: [
    {
      id: "1",
      content: "Expertise 1"
    },
    {
      id: "2",
      content: "Expertise 2"
    },
    {
      id: "3",
      content: "Expertise 3"
    },
    {
      id: "4",
      content: "Expertise 4"
    }
  ]
};

describe("ExpertiseSection", () => {
  afterEach(cleanup);

  it("Passes smoke test", () => {
    const { container } = render(<ExpertiseSection content={MOCK_DATA.items}/>);

    expect(container).toMatchSnapshot();
  });

  it("Contains expected children & content", () => {
    const { queryByTestId } = render(<ExpertiseSection content={MOCK_DATA.items}/>);

    expect(queryByTestId("expertiseList")).not.toBeNull();
    expect(queryByTestId("expertiseList").children.length).toBe(4);
    expect(queryByTestId(`expertiseList-item-1`).textContent).toBe("•   Expertise 1");
  });
});