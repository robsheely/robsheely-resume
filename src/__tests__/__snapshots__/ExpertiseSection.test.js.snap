// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertiseSection Passes smoke test 1`] = `
<div>
  <ul
    class="MuiGridList-root-3 ExpertiseSection-gridList-1"
    data-testid="expertiseList"
    id="expertiseList"
    style="margin: -2px;"
  >
    <p
      class="MuiTypography-root-4 MuiTypography-body1-13 ExpertiseSection-bullet-2"
      data-testid="expertiseList-item-1"
      id="expertiseList-item-1"
      style="width: 33.333333333333336%; height: 28px; padding: 2px;"
    >
      •   
      Expertise 1
    </p>
    <p
      class="MuiTypography-root-4 MuiTypography-body1-13 ExpertiseSection-bullet-2"
      data-testid="expertiseList-item-2"
      id="expertiseList-item-2"
      style="width: 33.333333333333336%; height: 28px; padding: 2px;"
    >
      •   
      Expertise 2
    </p>
    <p
      class="MuiTypography-root-4 MuiTypography-body1-13 ExpertiseSection-bullet-2"
      data-testid="expertiseList-item-3"
      id="expertiseList-item-3"
      style="width: 33.333333333333336%; height: 28px; padding: 2px;"
    >
      •   
      Expertise 3
    </p>
    <p
      class="MuiTypography-root-4 MuiTypography-body1-13 ExpertiseSection-bullet-2"
      data-testid="expertiseList-item-4"
      id="expertiseList-item-4"
      style="width: 33.333333333333336%; height: 28px; padding: 2px;"
    >
      •   
      Expertise 4
    </p>
  </ul>
</div>
`;
